src/test/test-label.c

```diff
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "label.h"
+#include "string-util.h"
+#include "tests.h"
+
+
+int pre_labelling_func(int dir_fd, const char *path, mode_t mode);
+int post_labelling_func(int dir_fd, const char *path);
+LabelOps test_label_ops = {
+        .pre = NULL,
+        .post = NULL
+    };
+
+TEST(label_ops_set) {
+
+    assert_se(label_ops_set(&test_label_ops) == 0);
+
+    /*attempt to reset label_ops when already set*/
+    assert_se(label_ops_set(&test_label_ops) == -EBUSY);
+
+}
+
+int pre_labelling_func(int dir_fd, const char *path, mode_t mode) {
+    /*sample pre_labelling task*/
+    if (dir_fd < 0 || !path || mode == 0)
+        return -1;
```

Comment:
do not make up errors like -1. Return negative errno style errors, i.e. -EINVAL or so. See coding style